lib/exercises.rb

```diff
@@ -1,19 +1,52 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
-
+# Time Complexity: O(n * m * log m) 
+	#	where n is the number of strings in the array to iterate over, 
+	# and m is the number of characters to sort in a string (m log m), 
+# Space Complexity: O(n * m) from sorting characters (m) and returning a new array (n)
+# where n is the number of strings in the array
+# and m is the number of characters in a string 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  sorted_strings_hash = Hash.new([])  
+
+  strings.each do |word|
+    word.chars.sort.join
+
+    if sorted_strings_hash[word]
+      sorted_strings_hash[word].push(word)
+		end
+	end
+
+	puts sorted_strings_hash.values
+	return sorted_strings_hash.values
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n + m log m) where n is the number of elements in the list and 
+	# and m is the number of unique elements to sort (m log m)
+# Space Complexity: O(m * k) where m is the number of unique elements to put in hash
+	# and k is the number of elements to return
 def top_k_frequent_elements(list, k)
```

Comment:
👍 I would say this is O(n log n) since k < n, and you're potentially sorting a hash of n elements. 

For space complexity since k < n, and you make a hash of potentially n elements this is O(n).